configuration/postgres/postgres.go

```diff
@@ -0,0 +1,432 @@
+		if item.Value != "" {
+			items[key] = &item
+		}
+	}
+	return &configuration.GetResponse{
+		Items: items,
+	}, nil
+}
+
+func (p *ConfigurationStore) Subscribe(ctx context.Context, req *configuration.SubscribeRequest, handler configuration.UpdateHandler) (string, error) {
+	var pgNotifyChannels []string
+	for k, v := range req.Metadata {
+		if res := strings.EqualFold(pgNotifyChannelKey, k) && !slices.Contains(pgNotifyChannels, v); res {
+			pgNotifyChannels = append(pgNotifyChannels, v) // append unique channel names only
+		}
+	}
+	if len(pgNotifyChannels) == 0 {
+		return "", fmt.Errorf("cannot subscribe to empty channel")
+	}
+	return p.subscribeToChannel(ctx, pgNotifyChannels, req, handler)
+}
+
+func (p *ConfigurationStore) Unsubscribe(ctx context.Context, req *configuration.UnsubscribeRequest) error {
+	p.configLock.Lock()
+	defer p.configLock.Unlock()
+	for _, v := range p.ActiveSubscriptions {
+		if v.uuid == req.ID {
+			break
+		} else {
+			return fmt.Errorf("unable to find subscription with ID : %v", req.ID)
+		}
+	}
+	if oldStopChan, ok := p.subscribeStopChanMap[req.ID]; ok {
+		delete(p.subscribeStopChanMap, req.ID)
+		close(oldStopChan.(chan struct{}))
+		for k, v := range p.ActiveSubscriptions {
+			if v.uuid == req.ID {
+				pgChannel := fmt.Sprintf(unlistenTemplate, k)
+				conn, err := p.client.Acquire(ctx)
+				if err != nil {
+					p.logger.Errorf("error acquiring connection:", err)
+					return fmt.Errorf("error acquiring connection: %s ", err)
+				}
+				defer conn.Release()
+				_, err = conn.Exec(ctx, pgChannel)
+				if err != nil {
+					p.logger.Errorf("error listening to channel:", err)
+					return fmt.Errorf("error listening to channel: %s", err)
+				}
+				delete(p.ActiveSubscriptions, k)
+			}
+		}
+		return nil
+	}
+	return fmt.Errorf("unable to find subscription with ID : %v", req.ID)
+}
+
+func (p *ConfigurationStore) doSubscribe(ctx context.Context, req *configuration.SubscribeRequest, handler configuration.UpdateHandler, command string, channel string, subscription string, stop chan struct{}) {
+	conn, err := p.client.Acquire(ctx)
+	if err != nil {
+		p.logger.Errorf("error acquiring connection:", err)
+		return
+	}
+	defer conn.Release()
+	if _, err = conn.Exec(ctx, command); err != nil {
+		p.logger.Errorf("error listening to channel:", err)
+		return
+	}
+	for {
+		notification, err := conn.Conn().WaitForNotification(ctx)
+		if err != nil {
+			if !pgconn.Timeout(err) && !errors.Is(ctx.Err(), context.Canceled) {
+				p.logger.Errorf("error waiting for notification:", err)
+			}
+			return
+		}
+		p.handleSubscribedChange(ctx, handler, notification, channel, subscription)
+	}
+}
+
+func (p *ConfigurationStore) handleSubscribedChange(ctx context.Context, handler configuration.UpdateHandler, msg *pgconn.Notification, channel string, subscriptionID string) {
+	defer func() {
+		if err := recover(); err != nil {
+			p.logger.Errorf("panic in handlesubscribedchange method and recovered: %s", err)
```

Comment:
Nitpick camelCase for method name...

Reply:
this method will is not part of interface and not visible outside the package. So it is named starting with a small case letter.